os/arch/arm/src/common/up_createstack.c

```diff
@@ -243,6 +252,18 @@ int up_create_stack(FAR struct tcb_s *tcb, size_t stack_size, uint8_t ttype)
 		heapinfo_exclude_stacksize(tcb->stack_alloc_ptr);
 #endif
 		board_led_on(LED_STACKCREATED);
+#ifdef CONFIG_MPU_STACKGUARD
+		/* The smallest size that can be programmed for an MPU region is 32 bytes */
+
+		tcb->stack_guard =	(uint32_t)tcb->stack_alloc_ptr + 0x40;
+		tcb->stack_guard = ((uint32_t)tcb->stack_guard & MPU_RBAR_ADDR_MASK);
+		tcb->guard_size = ((uint32_t)tcb->stack_guard - (uint32_t)tcb->stack_alloc_ptr);
```

Comment:
I think we need to update the tcb->adj_stack_size based on this calculation.

Reply:
updated